src/core/symbology/qgslinesymbollayer.h

```diff
@@ -651,6 +651,38 @@ class CORE_EXPORT QgsTemplatedLineSymbolLayerBase : public QgsLineSymbolLayer
      */
     void setPlacements( Qgis::MarkerLinePlacements placements ) { mPlacements = placements; }
 
+    /**
+     * Returns TRUE if the placement respects multi-part feature geometries.
+     *
+     * The default is FALSE, which means that Qgis::MarkerLinePlacement::FirstVertex or
+     * Qgis::MarkerLinePlacement::LastVertex placements will result in a symbol on
+     * the first/last vertex of EVERY part of a multipart feature.
+     *
+     * If TRUE, then Qgis::MarkerLinePlacement::FirstVertex or
+     * Qgis::MarkerLinePlacement::LastVertex placements will result in a symbol on
+     * the first/last vertex of the overall multipart geometry only.
+     *
+     * \see setRespectMultipart()
+     * \since QGIS 3.24
+     */
+    bool respectMultipart() const { return mRespectMultipart; }
```

Comment:
I'm not a super fan of this wording. What about `placeOnEveryPart` or something similar?